x-pack/plugins/ingest_manager/common/services/is_agent_upgradeable.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import semver from 'semver';
+import { Agent } from '../types';
+
+export function isAgentUpgradeable(agent: Agent, kibanaVersion: string) {
+  let agentVersion: string;
+  if (
+    typeof agent.local_metadata === 'object' &&
+    typeof agent.local_metadata.elastic === 'object' &&
+    typeof agent.local_metadata.elastic.agent === 'object' &&
+    typeof agent.local_metadata.elastic.agent.version === 'string'
```

Comment:
I think this also works

```suggestion
    typeof agent?.local_metadata?.elastic?.agent?.version === 'string'
```

Reply:
unfortunately it does not, i think because we don't type any of local_metadata possible attributes

Reply:
Hmm, interesting. What's the error?

Reply:
The error was this (which I find confusing): <img width="999" alt="Screen Shot 2020-10-01 at 7 46 10 PM" src="https://user-images.githubusercontent.com/1676003/94874001-c8d71180-041e-11eb-8ff5-f7847051d491.png">

But i've since made a type change where `local_metadata` is an object with keys of any: https://github.com/elastic/kibana/pull/78810/files#diff-66afa8d92013f1235fe54213c67f0583R114 .  I did this because they are not always strings.  Currently they sometimes point to booleans (local_metadata.elastic.agent.snapshot) or arrays of strings (`local_metadata.host.ip`).  Would like @nchaulet to approve this change, then the recommended change will work.

Reply:
Yes looks ok to me 👍 
Clearly not part of this PR but I am wondering if we could have stronger type for metadata, we are using Elastic ECS for them so maybe there is a typescript schema available somewhere

Reply:
@nchaulet it would be nice to have.  i've run into a similar issue with [index template settings](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-modules-settings).